src/main/java/seedu/apollo/command/module/ListModuleWithLessonCommand.java

```diff
@@ -0,0 +1,234 @@
+package seedu.apollo.command.module;
+
+import seedu.apollo.calendar.Calendar;
+import seedu.apollo.command.Command;
+import seedu.apollo.exception.module.InvalidModule;
+import seedu.apollo.exception.module.LessonNotAddedException;
+import seedu.apollo.exception.module.LessonTypeNotAddedException;
+import seedu.apollo.exception.utils.IllegalCommandException;
+import seedu.apollo.module.LessonType;
+import seedu.apollo.module.Module;
+import seedu.apollo.module.ModuleList;
+import seedu.apollo.module.Timetable;
+import seedu.apollo.storage.Storage;
+import seedu.apollo.task.TaskList;
+import seedu.apollo.ui.Ui;
+import seedu.apollo.utils.LoggerInterface;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.logging.ConsoleHandler;
+import java.util.logging.FileHandler;
+import java.util.logging.Level;
+import java.util.logging.LogManager;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import static seedu.apollo.utils.LessonTypeUtil.determineLessonType;
+
+public class ListModuleWithLessonCommand extends Command implements LoggerInterface {
+
+    private static Logger logger = Logger.getLogger("ListModuleCommand");
+    private String[] args;
+    private Module module;
+
+    /**
+     * Constructor for ListModuleWithLessonCommand.
+     *
+     * @param params     The parameters the user passes into the command.
+     * @param allModules The list of all modules.
+     * @throws InvalidModule If the module code is invalid.
+     */
+    public ListModuleWithLessonCommand(String params, ModuleList allModules) throws InvalidModule,
+            IllegalCommandException {
+
+        setUpLogger();
+        assert (params != null) : "ListModuleWithLessonCommand: ModuleCode should not be null!";
+
+        args = params.split("\\s+");
+
+        if (args.length != 1 && args.length != 2) {
+            throw new IllegalCommandException();
+        }
+
+        String moduleCode = args[0];
+
+        Module checkMod = allModules.findModule(moduleCode);
+        if (checkMod == null) {
+            throw new InvalidModule();
+        }
+
+        module = checkMod;
```

Comment:
can directly assign to module i think? ie module = allModules.findModule(moduleCode);